File: bubble_sort/go/bubble_sort.go
package main

import (
	"fmt"
	"time"

	utils "github.com/ggwellplayed4101/Algorithms/list_generator"
)

const size int = 1000

func main() {
	unsorted_list := utils.Random_Arr(size)

	start := time.Now()
	bubble_sort(unsorted_list)
	elapsed := time.Since(start).Seconds()

	fmt.Println("Execution time:", elapsed, "seconds")

}

func bubble_sort(list []int) []int {
	length := len(list)
	for i := 0; i < length-1; i++ {
		for j := 0; j < length-1-i; j++ {
			if list[j] > list[j+1] {
				var temp int = list[j+1]
				list[j+1] = list[j]
				list[j] = temp
			}
		}
	}
	return list

}

File: bubble_sort/python/bubble_sort.py
import time
from list_generator import list_generator


def main():
    unsorted_list = list_generator()

    start = time.time()
    sorted_list = bubble_sort(unsorted_list)
    end = time.time()

    elapsed_time = end - start

    print(f"Time taken by bubble_sort: {elapsed_time:.7f} seconds")

def bubble_sort(list):
    for j in range(len(list) - 1):
        for i in range(len(list) - 1):
            if list[i] > list[i+1]:
                temp = list[i]
                list[i] = list[i+1]
                list[i+1] = temp
    return list

if __name__ == "__main__":
    main()